### PR TITLE
[exporter/datadog] Add `host_metadata` section

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:**

Add `host_metadata` section to group related settings.

**Link to tracking Issue:** #9099

**Testing:** 

- [x] Added unit tests
- [x] Check that deprecation warnings are logged
- [x] Test that host metadata keeps being sent when `only_metadata` is false (not covered by tests)

**Documentation:** Updated documentation

### CODE REVIEW

File: exporter/datadogexporter/config/config.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 
 var (
 	errUnsetAPIKey = errors.New("api.key is not set")
-	errNoMetadata  = errors.New("only_metadata can't be enabled when send_metadata or use_resource_metadata is disabled")
+	errNoMetadata  = errors.New("only_metadata can't be enabled when host_metadata::enabled = false or hostname_source != first_resource")
```
Comment (albertvaka) [inline exporter/datadogexporter/config/config.go:34]:
```suggestion
	errNoMetadata  = errors.New("only_metadata can't be enabled when host_metadata::enabled = false or host_metadata::hostname_source != first_resource")
```

**Review by albertvaka — APPROVED** [review]

**Comment by jpkrohling** [comment]
I trust @mx-psi, but would it be possible to get an approval from another code owner as well? Like @KSerrania, @gbbr or @knusbaum

**Comment by mx-psi** [comment]
@jpkrohling Sure, I will ask internally for this

**Review by KSerrania — APPROVED** [review]
Left a few suggestions about comments, overall LGTM

File: exporter/datadogexporter/config/config.go [hunk 2]
```diff
@@ -281,18 +340,22 @@ type Config struct {
...
 	SendMetadata bool `mapstructure:"send_metadata"`
 
 	// OnlyMetadata defines whether to only send metadata
 	// This is useful for agent-collector setups, so that
 	// metadata about a host is sent to the backend even
 	// when telemetry data is reported via a different host.
 	//
-	// This flag is incompatible with disabling `send_metadata`
+	// This flag is incompatible with disabling host metadata
 	// or `use_resource_metadata`.
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply exporter/datadogexporter/config/config.go]: > Btw, why do we set only_metadata as incompatible with host_metadata::hostname_source != first_resource?

I don't think this restriction makes a lot of sense, but I would rather think about it a bit more and change this in a different PR

**Review by KSerrania — APPROVED** [review]

**Comment by mx-psi** [comment]
This is ready to merge, but I am babysitting the CI until I finally get a test run without flakes

**Review by jpkrohling — COMMENTED** [review]
I'm approving on behalf of the other reviews, done by code owners.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**